packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -2107,7 +2108,7 @@ export class MergeTree {
 
 		this.slideAckedRemovedSegmentReferences(localOverlapWithRefs);
 		// opArgs == undefined => test code
-		if (movedSegments.length > 0) {
```

Comment:
i think this might need to be a bit more complicated, in we don't want to event on remote operations that affect no segments.

Reply:
I'd push back on that. We event on remote operations that have tiny effects on the merge tree state. A zero-length obliterate is still a delta of the merge tree state, even though it does not affect any segments. Distinguishing between local and remote obliterates just makes the contract of `mergeTreeDeltaCallback` really complicated.

It's better to air on calling the callback a bit too much than to assume that no one would need it and then find out otherwise later (which is what caused the need for this PR in the first place).

Reply:
we've received feedback from multiple partners that they don't like empty events, which is why it works that way. if nothing is affected there is nothing the application needs to do.

Reply:
As discussed in standup, we will emit the event from mergeTree because we need it in sequence, but we can choose to not emit a sequenceDelta event if no segments are changed.